scrabble.js

```diff
@@ -1,14 +1,133 @@
-
 const Scrabble = {
+
   score(word) {
+    let wordArray = word.split('')
+    let length = wordArray.length
+    let sum = 0;
+
+    if (length === 7) {
+      sum = 50;
+    } else if (length > 7)  {
+      throw 'Word length must be less than 7';
+    } else if (length === 0) {
+      throw 'Word cannot be empty';
+    }
+
+    if (/[*&^$%#@!?]/.test(word)) {
+      throw 'Word cannot have symbols';
+    }
 
+    for( let letter of wordArray ) {
+      switch (true) {
+        case /[aeioulnrst]/i.test(letter):
+        sum += 1;
```

Comment:
I'm not a big fan of organizing the letter scores in a `switch` statement like this. When the data is intertwined with the code that uses it, there's no easy way to reuse this information.

For example, what if the player had a hand full of tiles, and you wanted to be able to display the value of each of those letters? You would have to copy/paste this switch in that code, or at least extract it to a method.

A better approach would be to store the scores in a hash:

```javascript
const LETTER_VALUES = {
  a: 1,
  b: 3,
  // ...
}
```

then here you could use it like this:

```javascript
word.forEach((letter) => {
  const value = LETTER_VALUES[letter.toLowerCase()];
  if (value) {
    totalScore += letter.toLowerCase();
  } else {
    throw new Error(`Invalid character ${letter}`);
  }
}
```